File: src/main/java/com/xebia/xboson/TagSelectionRule.java
package com.xebia.xboson;

import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.util.Optional;

/**
 * Created by xviuda on 22-01-16.
 */
public class TagSelectionRule implements SelectionRule {
    @Override
    public Optional<ParsedElement> match(Document doc, Element element) {
        String tagName = element.tagName();
        Elements found = doc.select(tagName);
        if (found.size() == 1) {
            return Optional.of(new ParsedElement(tagName, tagName));
        }else {
            return Optional.empty();
        }
    }
}

File: src/main/kotlin/com/xebia/xboson/TagSelectionRule.kt
package com.xebia.xboson

import org.jsoup.nodes.Document
import org.jsoup.nodes.Element
import org.jsoup.select.Elements

import java.util.Optional

class TagSelectionRule : SelectionRule {
    override fun match(document: Document, element: Element): ParsedElement? {
        val tagName = element.tagName()
        val found = document.select(tagName)
        if (found.size == 1) {
            return ParsedElement(tagName, tagName)
        } else {
            return null
        }
    }
}
